ns-app/scratch/webrtc_test/trace_player.cc

```diff
@@ -74,16 +81,27 @@ void TracePlayer::PlayTrace(size_t trace_index) {
             auto device =
                 dynamic_cast<PointToPointNetDevice *>(PeekPointer(node->GetDevice(j)));
             if (device) {
-                device->SetDataRate(DataRate(trace.capacity_ * 1e3));
                 // set loss rate in every device
                 if (trace.loss_rate_) {
-                    Ptr<RateErrorModel> em = CreateObjectWithAttributes<RateErrorModel> ("RanVar", StringValue("ns3::UniformRandomVariable[Min=0.0|Max=1.0]"), \
-                                                                                     "ErrorRate", DoubleValue (trace.loss_rate_.value()), \
-                                                                                     "ErrorUnit", StringValue("ERROR_UNIT_PACKET"));
-                    device->SetAttribute("ReceiveErrorModel", PointerValue (em));
+                    SetLossRate(device, trace.loss_rate_.value());
+                }
+                if (trace.capacity_ == 0) {
```

Comment:
The type of "capacity_" is double.
To comparing with other numbers, there may be precision issues.

Reply:
The capacity_ is uint64 in our system, I want to adapt some trace format that capacity_ is `double`, so for `lexical_cast` I use `double` to cast it. 

https://github.com/OpenNetLab/gym/blob/72dd6d9cc8b43d61517371e015b0325f1345abf8/ns-app/scratch/webrtc_test/trace_player.h#L11

of course, this conversion will  loss some precision, but I think the tiny precision isn't significant in our system.